alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -180,31 +116,15 @@ void BasicBehaviour::doTrigger()
     _engine->editScheduler().schedule(std::bind(&BasicBehaviour::runJob, this, nullptr));
 }
 
-void BasicBehaviour::terminateJob()
+void BasicBehaviour::doTerminate()
 {
-    // There will be no run job if the interval is 0 or init is not executed
-    if (_activeRunJobId != -1) {
-        _engine->editScheduler().cancelJob(_activeRunJobId);
-        _activeRunJobId = -1;
-    }
+    setTerminatedState();
     // Just to be double safe in terms of the correct behaviour of isSuccess() & isFailure() ensure result is reset before incrementing _execState
     _behResult.store(BehResult::UNKNOWN);
-    ++_execState;
```

Comment:
as the comment mentions incrementing _execState before clearing the result could lead to isSuccess() & isFailure() returning true even though the behaviour has never executed in the context of the current running plan i.e. _execState++ can make it equal to _signalState & so isExecutingInContext will be true & consequently isSuccess() & isFailure() can be true & yet the behaviour has not even been initialised in the _signalState's context. So we return the success mark from the previous run which is a problem.